domain/relation/types.go

```diff
@@ -25,11 +26,29 @@ type GetRelationEndpointUUIDArgs struct {
 
 // RelationDetails represents the current application's view of a relation.
 type RelationDetails struct {
-	Life     life.Value
-	UUID     corerelation.UUID
-	ID       int
-	Key      string
-	Endpoint []Endpoint
+	// Life is the current life value of the relation.
+	Life life.Value
+	// UUID is the unique identifier of the relation.
+	UUID corerelation.UUID
+	// ID is the sequential ID of the relation.
+	ID int
+	// Key is the natural key of the relation.
+	Key corerelation.Key
+	// Endpoints are the endpoints of the relation.
+	Endpoints []relation.Endpoint
```

Comment:
question: Why is this change made? They are the same.

The change has no indication of why it was made in this PR nor the commit message. 

Prefer the domain rather than internal.

Reply:
I made this change because this was the only use of the Endpoint struct in the domain, and the two types are identical. I get that we should prefer creating types in the domain, but I wanted to just have one consistent version everywhere.

I've changed it back.

Reply:
Right now the domain is getting messy with some methods using the internal Endpoint and some using the domain Endpoint. It'd it a lot nicer just to have one, shall I bring a PR to move everything to just using the domain Endpoint struct and move the methods across too?